### ISSUE TITLE
Operations: merge

### ISSUE THREAD

**Issue by tmcw** [openstreetmap/iD#435]
This should be a 'unified merge' I think, which works for (uses different algorithms) for lines & for areas, but has the same UI.

**Comment by homersimpsons** [comment]
Sorry for #3074 didn't find this tread when research...
I hope this will exist, in France a lot of building are extracted from 'cadastre' (government map) but there are a lot of buildings composed of multiple area like [here (google maps)](https://www.google.fr/maps/@44.6656798,2.1594132,20.6z) , I have corrected this in [OSM](http://www.openstreetmap.org/#map=19/44.66561/2.16006) [I kept some fragments because I'm not there to check if it's really one building or differents ones]. But it's ugly and constitute a lack of space. This tool could be apply on 70% of multiple area I think, so this would be really usefull.
Another helpfull tool could be split (from 2 non consecutive points of an area) in 2 area, this is usefull when mapping adress a group of building.

**Comment by egonzale** [comment]
Hi, I think this is the right place to mention a problem we had yesterday.
Mapping riverbanks as indicated in wiki, we make them by sections, so we draw polygons that have a common border where they continue (see example [here](http://www.openstreetmap.org/edit?editor=id&#map=18/10.22995/105.35048) ).
Problem is that if you select them and merge them (using the + symbol), it merges "ways" instead of "area" which is what you would expect... mergin these areas as ways simply breaks it all (you can test in iD and then try to make some sense of it in JOSM...)
Would it make sense to change the algorithm when two areas are selected (and have a common border) they are merged into one single area? Does it make sense?
As reference, the action currently implemented in the + symbol seems to do the same as in JOSM Tools>Combine ways ... the one I would expect is JOSM Tools>Join overlaping areas

**Comment by homersimpsons** [comment]
@egonzale This is exactly what I was asking for in my previous answer :) Hope this will be realised.

**Comment by bhousel** [comment]
Hi @egonzale and @homersimpsons, yes I'd really like to see iD have an area merge tool.  Ideally my plan is to bring in turf's polygonal operations like `turf-union`, `turf-difference`, `turf-intersect` after they finish removing the JSTS dependency, see https://github.com/Turfjs/turf/issues/88

This is something that the Turf team could use help with and would benefit not just iD but a lot of projects. Maybe someone whose skills lie at the intersection of JavaScript and computational geometry could move this across the finish line (paging dr. @paulmach?).

**Comment by EdLoach** [comment]
If there is a more relevant open issue I've not found it, but apparently it is possible to merge a split roundabout back into a single way without even a warning about the 132 route relations that pass through the roundabout.

**Comment by quincylvania** [comment]
@EdLoach That sounds like something more specific than this. Feel free to open a new issue for it.

**Comment by Filip10** [comment]
Are there some news about merging two areas into one area (not multipolygon)?

A lot of people are making not needed multipolygons instead of one area.

Maybe recommend them using JOSM and function "Join overplapping areas" (Shift+J)? ;)

**Comment by dbdean** [comment]
I agree with @Filip10's comments. I wanted to merge two adjacent shops recently that had been combined into one (in the real world), and it took me a while to realise it was creating a multipolygon. I would have expected it to just join the two areas. I ended up having to go and do it in JOSM.

**Comment by Binnette** [comment]
> Are there some news about merging two areas into one area (not multipolygon)?
> 
> A lot of people are making not needed multipolygons instead of one area.
> 
> Maybe recommend them using JOSM and function "Join overplapping areas" (Shift+J)? ;)

Same here. I would love to see an option to "really merge two areas" and not create a multipolygon with them.

I am pretty sure it was working "correctly" before in ideditor. This feature is really essential for me.

Have a nice day :+1:

**Comment by hb0nd** [comment]
what is needed for this to be implemented? it's a really sorely missing feature imo
